### PR TITLE
Feature: added as gen custom types to proto cli

### PR DESCRIPTION
**Pull request by 0xSwapFeeder**

### CODE REVIEW

**Review by Elli610 — CHANGES_REQUESTED** [review]
conflicts need to be fixed

**Comment by 0xSwapFeeder** [comment]
Unable to make it work for now, on hold

**Comment by 0xSwapFeeder** [comment]
Rebased and working, pls test on your side before merging to avoid the "it works on my machine" syndrome!

**Pull request merged**
